hazelcast-sql/src/test/java/com/hazelcast/sql/SqlErrorClientTest.java

```diff
@@ -296,11 +296,14 @@ public void testRowError_deserialization() {
             try (SqlResult result = client.getSql().execute("SELECT this FROM " + MAP_NAME)) {
                 boolean first = true;
 
-                for (SqlRow ignore : result) {
+                for (SqlRow row : result) {
                     if (first) {
                         BadValue.READ_ERROR.set(true);
 
                         first = false;
+                    } else {
+                        // row values are deserialized lazily, on access
+                        row.getObject("this");
```

Comment:
We should also select `__key` and ensure that we can read it successfully.

This might be more readable:
```java
Iterator it = result.iterator();
SqlRow row = iterator.next();
row.getObject("this");
SqlRow row = iterator.next();
row.getObject("__key"); // suceeds
assertThatThrownBy(() -> row.getObject("this"));
```